cmd/tuf/server/main.go

```diff
@@ -140,26 +149,112 @@ func main() {
 	// worries here.
 	var compressed bytes.Buffer
 	if err := repo.CompressFS(os.DirFS(dir), &compressed, map[string]bool{"keys": true, "staged": true}); err != nil {
-		logging.FromContext(ctx).Fatalf("Failed to compress the repo: %v", err)
+		return fmt.Errorf("failed to compress the repo: %v", err)
 	}
 	data["repository"] = compressed.Bytes()
 
 	if !*noK8s {
 		nsSecret := clientset.CoreV1().Secrets(ns)
-		if err := secret.ReconcileSecret(ctx, *secretName, ns, data, nsSecret); err != nil {
-			logging.FromContext(ctx).Panicf("Failed to reconcile secret %s/%s: %v", ns, *secretName, err)
+		if err := secret.ReconcileSecret(ctx, repoSecretName, ns, data, nsSecret); err != nil {
+			return fmt.Errorf("failed to reconcile secret %s/%s: %v", ns, repoSecretName, err)
+		}
+
+		// If we should also store created keys in a secret, read all their files and save them in the secret
+		if keysSecretName != "" {
+			keyFiles, err := os.ReadDir(filepath.Join(dir, "keys"))
+			if err != nil {
+				return fmt.Errorf("failed to list keys directory %v", err)
+			}
+			dataKeys := map[string][]byte{}
+			for _, keyFile := range keyFiles {
+				if !strings.HasSuffix(keyFile.Name(), ".json") {
+					continue
+				}
+				keyFilePath := filepath.Join(filepath.Join(dir, "keys", keyFile.Name()))
+				content, err := os.ReadFile(keyFilePath)
+				if err != nil {
+					return fmt.Errorf("failed reading file %s: %v", keyFilePath, err)
+				}
+				dataKeys[keyFile.Name()] = content
+			}
+			if err := secret.ReconcileSecret(ctx, keysSecretName, ns, dataKeys, nsSecret); err != nil {
+				return fmt.Errorf("failed to reconcile keys secret %s/%s: %v", ns, keysSecretName, err)
+			}
+		}
+	}
+
+	logging.FromContext(ctx).Infof("tuf repository was created in: %s, moving to %s", dir, targetDir)
+
+	// Copy repository to the targetDir - until Go 1.23 which has os.CopyFS, we use
+	// a quick hack where we uncompress the compressed repository to the targetDir
+	repo.Uncompress(bytes.NewReader(data["repository"]), targetDir)
```

Comment:
Question: Right now, it seems like we're creating in a temp directory and then moving to the final destination.  What about having the `repo.CreateRepo` take `targetDir` as an argument, and not needing to do this unpack?

At that point, the code could also look like:

```golang
if noK8s {
  // We're done
  return nil
}
```

I suppose one advantage of the copy approach is that you won't end up with a half-initialized directory if you crash partway through....

Reply:
There were two reasons why I didn't add the extra argument, even though we considered it:

* Exactly as you said, I think it's good to not end up with a potentially half-initialized directory.
* Adding the new argument would change the public API of `repo.CreateRepo`. While I think that the modules from this repo might not be depended on very much, I wanted to prevent API breakage (but actually talking about this right now, I'm curious if that's ok - for future contributions).